test/modules/CloudToDeviceMessageTester/CloudToDeviceMessageSender.cs

```diff
@@ -58,36 +58,37 @@ public async Task StartAsync(CancellationToken ct)
 
             while (!ct.IsCancellationRequested && this.IsTestTimeUp(testStartAt))
             {
-                MessageTestResult testResult = await this.SendCloudToDeviceMessageAsync(batchId, this.trackingId);
-                await ModuleUtil.ReportTestResultAsync(this.testResultReportingClient, this.logger, testResult);
-                this.messageCount++;
-                await Task.Delay(this.messageDelay, ct);
+                MessageTestResult testResult = null;
+                try
+                {
+                    testResult = await this.SendCloudToDeviceMessageAsync(batchId, this.trackingId);
+                    this.messageCount++;
+                    await ModuleUtil.ReportTestResultAsync(this.testResultReportingClient, this.logger, testResult);
+                    await Task.Delay(this.messageDelay, ct);
+                }
+                catch (Exception ex)
+                {
+                    this.logger.LogInformation($"Error occurred while sending Cloud to Device message for Sequence number: {this.messageCount}, batchId: {batchId}. Error message: {ex.Message}");
+                }
             }
         }
 
         internal async Task<MessageTestResult> SendCloudToDeviceMessageAsync(Guid batchId, string trackingId)
         {
             this.logger.LogInformation($"Sending C2D message to deviceId: {this.deviceId} with Sequence Number: {this.messageCount}, batchId: {batchId}, and trackingId: {trackingId}");
-            try
-            {
-                var message = new Message(Encoding.ASCII.GetBytes("Cloud to device message."));
-                message.Properties.Add(TestConstants.Message.SequenceNumberPropertyName, this.messageCount.ToString());
-                message.Properties.Add(TestConstants.Message.BatchIdPropertyName, batchId.ToString());
-                message.Properties.Add(TestConstants.Message.TrackingIdPropertyName, trackingId);
-                await this.serviceClient.SendAsync(this.deviceId, message);
 
-                return new MessageTestResult(this.moduleId + ".send", DateTime.UtcNow)
-                {
-                    TrackingId = trackingId,
-                    BatchId = batchId.ToString(),
-                    SequenceNumber = this.messageCount.ToString()
-                };
-            }
-            catch (Exception ex)
+            var message = new Message(Encoding.ASCII.GetBytes("Cloud to device message."));
+            message.Properties.Add(TestConstants.Message.SequenceNumberPropertyName, this.messageCount.ToString());
+            message.Properties.Add(TestConstants.Message.BatchIdPropertyName, batchId.ToString());
+            message.Properties.Add(TestConstants.Message.TrackingIdPropertyName, trackingId);
+            await this.serviceClient.SendAsync(this.deviceId, message);
```

Comment:
Why did you move the try-catch() up the call stack?  (Pure curiosity).. 
I'd expect the exception to be thrown by `await this.serviceClient.SendAsync(this.deviceId, message);` and you can prob just wrap the try-catch() here like in your original code.

Reply:
The way I had it originally, if I catch the DeviceExceededMaximum (legal in this case), then I'll also update the sequence number, which I don't want to do, since I want to retry on that sequence number until it goes through. 

Now, if the Send method throws the DeviceExceededMaximum exception, we will restart the whole call, and we can update the sequence number only once the send goes through